packages/@n8n/ai-workflow-builder.ee/src/workflow-builder-agent.ts

```diff
@@ -181,6 +183,18 @@ export class WorkflowBuilderAgent {
 	}
 
 	async *chat(payload: ChatPayload, userId?: string) {
+		// Check for the message maximum length
+		if (payload.message.length > MAX_AI_BUILDER_PROMPT_LENGTH) {
+			this.logger?.warn('Message exceeds maximum length', {
+				messageLength: payload.message.length,
+				maxLength: MAX_AI_BUILDER_PROMPT_LENGTH,
+			});
+
+			throw new Error(
```

Comment:
Can we use `ApplicationError` here to add context if it gets reported to Sentry?

Reply:
There's also `ValidationError` in `packages/@n8n/ai-workflow-builder.ee/src/errors/index.ts` that would be fitting here. But we probably shouldn't report these to sentry? 🤔